src/libraries/Common/src/Interop/OSX/System.Native/Interop.SearchPath.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_SearchPath")]
+        internal static extern string SearchPath(NSSearchPathDirectory folderId);
```

Comment:
It's this can return null, it should be `string?`.  Same for other places or might propagate.

Reply:
@stephentoub should I also add `#nullability enable` ?

Reply:
Wonder if ` Interlocked.CompareExchange(ref s_specialFolders, new Dictionary<SpecialFolder, string>(), null);` tells Roslyn it's not null anymore

Reply:
> should I also add #nullability enable ?

You shouldn't need to.
https://github.com/dotnet/runtime/blob/be469ad6b432fe10156cfbbc13ddc67b78a91024/src/mono/netcore/System.Private.CoreLib/System.Private.CoreLib.csproj#L52

> Wonder if Interlocked.CompareExchange(ref s_specialFolders, new Dictionary<SpecialFolder, string>(), null); tells Roslyn it's not null anymore

That will inform the compiler that, immediately after this call, s_specialFolders isn't null.

Unfortunately it looks like mono's Corelib.csproj is disabling all nullability warnings:
https://github.com/dotnet/runtime/blob/be469ad6b432fe10156cfbbc13ddc67b78a91024/src/mono/netcore/System.Private.CoreLib/System.Private.CoreLib.csproj#L123-L125
I'll take a look at fixing that.